ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -499,56 +495,35 @@ private byte[] serializeValue(
    * The best option was to dynamically look at the SR schema during an INSERT statement.
    */
   private static FormatInfo addSerializerMissingFormatFields(
-      final SchemaRegistryClient srClient,
       final FormatInfo formatInfo,
       final String topicName,
       final boolean isKey
   ) {
-    // Only SR information, such as schema ids and full names are required, which are available
-    // for SR formats.
+    // Just add missing fields required for serialization SR formats
     final Format format = FormatFactory.fromName(formatInfo.getFormat());
     if (!format.supportsFeature(SerdeFeature.SCHEMA_INFERENCE)) {
       return formatInfo;
     }
 
-    final Set<String> supportedProperties = format.getSupportedProperties();
+    // If SCHEMA_ID is not specified, then add the SUBJECT_NAME which helps the serializer
+    // to identify the schema to fetch from SR but without using the restrictions we
+    // have with SCHEMA_ID
+    if (!formatInfo.getProperties().containsKey(ConnectProperties.SCHEMA_ID)) {
+      final Set<String> supportedProperties = format.getSupportedProperties();
 
-    final ImmutableMap.Builder propertiesBuilder = ImmutableMap.builder();
-    propertiesBuilder.putAll(formatInfo.getProperties());
+      // Add SUBJECT_NAME only on supported SR formats
+      if (supportedProperties.contains(ConnectProperties.SUBJECT_NAME)) {
+        final ImmutableMap.Builder propertiesBuilder = ImmutableMap.builder();
+        propertiesBuilder.putAll(formatInfo.getProperties());
 
-    // The FULL_SCHEMA_NAME allows the serializer to choose the schema definition
-    if (supportedProperties.contains(ConnectProperties.FULL_SCHEMA_NAME)) {
-      if (!formatInfo.getProperties().containsKey(ConnectProperties.FULL_SCHEMA_NAME)) {
-        SchemaRegistryUtil.getLatestParsedSchema(srClient, topicName, isKey).map(ParsedSchema::name)
-            .ifPresent(schemaName ->
-                propertiesBuilder.put(ConnectProperties.FULL_SCHEMA_NAME, schemaName));
-      }
-    }
+        propertiesBuilder.put(ConnectProperties.SUBJECT_NAME,
```

Comment:
Why do we need to pass the subject name here? It looks like downstream it's used in order to get a schema ID -- why can't we just get the schema ID from the subject name here and then pass the schema ID, in order to avoid introducing a new property?

Reply:
It was like you suggests in a another PR, but Hao recommended to not inject the SCHEMA_ID if it was not explicitly specified in the WITH properties. The reason is that in the ProtobufSerdeFactory, the SCHEMA_ID is used to configure the converter (See https://github.com/confluentinc/ksql/blob/master/ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufSerdeFactory.java#L179). The serialization then takes a different path internally (in the SR API) when that config is set, which causes some corner cases to fail during INSERT VALUES. 

For instance, the 'auto.register.schemas' is disabled when SCHEMA_ID is present. If we set this ID on any case, then the auto.register.schemas will be false for any case, which will not register a missing schema in SR during INSERT VALUES. This is an issue I raised, but we haven't discussed if we want to keep it or not. So I ended up passing a SUBJECT_NAME which will keep the current behavior. It was that, or a boolean value in the Format class to say `do not set the ID in the converter config`. I don't like this SUBJECT_NAME approach, but it was the change with less refactor code. The other option was to pass the topic name on the createSerde() so it creates the subject name in the serde, but that refactor causes too many files to change.

Reply:
> If we set this ID on any case, then the auto.register.schemas will be false for any case, which will not register a missing schema in SR during INSERT VALUES. This is an issue I raised, but we haven't discussed if we want to keep it or not.

My vote is to remove the behavior (i.e., I think we should disable auto-registering of schemas) if it means our code can be a lot cleaner, which it sounds like it does.

This SUBJECT_NAME approach doesn't feel right to me since subject name doesn't seem like it should be considered a format property (I don't think schema ID really is, either, but we already had this conversation when schema ID was first introduced), and it's confusing that subject name is only used as an internal format property (users can't set it) whereas most of the other properties can be set by users.

Is the strategy for cleaning up this code (assuming we get rid of auto registering schemas) clear to you? I'm fine with this SUBJECT_NAME approach as a short-term solution if so. I really don't think we should keep this on a medium-term, though. This code is really hard to reason about.

Can we document (in code comments) that the SUBJECT_NAME property is internal-only, and is safe to remove in the future because it's only ever added during INSERT VALUES which means it's never written to the command topic? If it's protobuf-only (looks like it is), we can also move it just to the ProtobufProperties class instead of ConnectProperties so that it's even less likely someone will accidentally wire it into somewhere we don't want it.